### PR TITLE
Build #9963 [v98] New automation to detect and build the app with rus…

### LINKED ISSUE #9963: Modify automation to detect new rust component version
Before SPM work we had a github action and automation in place to detect new A-S version and modify cartfile files. Now we need to modify that so that we update the Package.resolved and the proper files to detect and build using the new rust component version.
Something similar is already working fine on Focus-iOS: https://github.com/mozilla-mobile/focus-ios/actions/workflows/check-rust-component-dependency.yml

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-3726)

### PR DESCRIPTION
**Pull request by isabelrios**
…t component newer version

PR to fix #9963

### CODE REVIEW

File: test-fixtures/update-rust-component-version.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    latest_tag = repo.get_tags()[0].name
+    latest_commit = str(repo.get_tags()[0].commit)
+    only_commit = re.findall(r'"([^"]*)"', latest_commit)
+
+    return (str(latest_tag), str(only_commit[0]))
+
+def read_rust_components_tag_version():
+    # Read Package file to find the current rust-component version
+    f = open(SPM_PACKAGE)
+    data = json.load(f)
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:25]: `try-catch`

File: test-fixtures/update-rust-component-version.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    repo = g.get_repo(GITHUB_REPO)
+
+    latest_tag = repo.get_tags()[0].name
+    latest_commit = str(repo.get_tags()[0].commit)
+    only_commit = re.findall(r'"([^"]*)"', latest_commit)
+
+    return (str(latest_tag), str(only_commit[0]))
+
+def read_rust_components_tag_version():
+    # Read Package file to find the current rust-component version
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:23]: Current or newest?

File: test-fixtures/update-rust-component-version.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    f = open(SPM_PACKAGE)
+    data = json.load(f)
+    pin = data["object"]
+
+    for i in pin["pins"]:
+        if i["package"] == "MozillaRustComponentsSwift":
+            # return the json with the RustComponent info
+            json_new_version = i["state"]
+    f.close()
+    # Return the current version and commit
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:33]: See above

File: test-fixtures/update-rust-component-version.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+                version_found = minimumVersion.find("=")
+                last_line_position = minimumVersion.find(";")
+                current_tag_version = ''
+                # version format: XX.Y.Z
+                for i in range(version_found+2 , last_line_position):
+                    current_tag_version+= minimumVersion[i]
+                # Return the current rust-component version in project
+                return current_tag_version
+
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:60]: This looks super fragile 😨

File: test-fixtures/update-rust-component-version.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+def compare_versions(current_tag_version, repo_tag_version):
+    # Compare rust-component version used and the latest available in rust-component repo
+    if current_tag_version < repo_tag_version:
+        print("Update rust componet version and create PR")
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:65]: nit: typo, `Update Rust Component version and create a PR`

File: test-fixtures/update-rust-component-version.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        return True
+    else:
+        print("No new versions, skip")
+        return False
+
+def update_spm_file(current_tag, current_commit, rust_component_repo_tag, rust_component_repo_commit, file_name):
+    # Read the SPM package file and update it
+    file = open(file_name, "r+")
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:73]: `try-catch`

File: test-fixtures/update-rust-component-version.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    data = data.replace(current_commit, rust_component_repo_commit)
+    file.close()
+    
+    file = open(file_name, "wt")
+    file.write(data)
+    file.close()
+
+def update_proj_file(current_tag, rust_component_repo_tag, file_name):
+    file = open(file_name, "r+")
+    data = file.read()
```
Comment (AaronMT) [inline test-fixtures/update-rust-component-version.py:85]: same in this function

**Review by AaronMT — CHANGES_REQUESTED** [review]

File: test-fixtures/update-rust-component-version.py (hunk 2 above)
Comment (isabelrios) [reply test-fixtures/update-rust-component-version.py:23]: Current as it is reading the rust-component version in project file

File: test-fixtures/update-rust-component-version.py (hunk 4 above)
Comment (isabelrios) [reply test-fixtures/update-rust-component-version.py:60]: I would like to improve it, but I can't find another way to go throught the file and read it.. I has been working previously on the A-S implementation and now in focus-iOS doing same thing

**Comment by isabelrios** [comment]
@AaronMT thanks for your review, I did the requested changes although I'm not very sure about the best way to add the try-catch, let me know if they way I added them is right. The code is working correctly but I guess there is a standard way to write them.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**